samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c

```diff
@@ -433,6 +446,9 @@ static void cloud_event_handler(const struct nrf_cloud_evt *nrf_cloud_evt)
 		 * when devices are first associated with an nRF Cloud account.
 		 */
 		atomic_set(&initial_association, true);
+		LOG_DBG("Clearing CLOUD_READY");
+		k_event_clear(&cloud_events, CLOUD_READY);
+		start_readiness_timeout(CONFIG_CLOUD_ASSOCIATION_TIMEOUT_SECONDS);
```

Comment:
Is this needed? It complicates the control flow (now the readiness timer is no longer strictly coupled to the connection loop), and the device should not ever get into ready state until after `NRF_CLOUD_EVT_READY` anyways, correct?

Reply:
It's needed if I don't follow your other suggestion to move the ping into the library.

Reply:
Cloud ready needs to be cleared because it is not, in fact, ready. The sample cannot successfully perform any nRF Cloud operation until the device is associated.

Reply:
I more meant, won't the `CLOUD_READY` flag already be false by this stage in connection, since presumably `NRF_CLOUD_EVT_READY` isn't called?

Or am I mistaken about that?

Although as you say it's moot if you run with my other suggestions, so if you're planning to do that, I'm happy to let this discussion drop 😆

Reply:
Clearing `CLOUD_READY` is needed in the case that while the device is connected the user deletes it from their account. In that case, the sample keeps acting like everything is fine, but it isn't.

This is regardless of whether I move the ping inside the library or not.

Reply:
If the device is deleted from the account, doesn't this trigger a disconnect which would clear the `CLOUD_READY` flag?

Reply:
I needed it at some point during development of this PR, but appear to not need it now.